fpm/src/fpm_compiler.f90

```diff
@@ -46,206 +25,302 @@ subroutine add_compile_flag_defaults(build_name,compiler,model)
-       & -align all&
-       & -error-limit 1&
-       & -reentrancy threaded&
-       & -nogen-interfaces&
-       & -assume byterecl&
-       &'
-       mandatory=' -module '//modpath//' -I '//modpath 
-    case('debug_ifort')
-       fflags = '&
-       & -warn all&
-       & -check:all:noarg_temp_created&
-       & -error-limit 1&
-       & -O0&
-       & -g&
-       & -assume byterecl&
-       & -traceback&
-       &'
-       mandatory=' -module '//modpath//' -I '//modpath 
-    case('release_ifx')
-       fflags = ' '
-       mandatory=' -module '//modpath//' -I '//modpath 
-    case('debug_ifx')
-       fflags = ' '
-       mandatory=' -module '//modpath//' -I '//modpath 
-
-    case('release_pgfortran','release_pgf90','release_pgf95')  ! Portland Group F90/F95 compilers
-       fflags = ' '
-       mandatory=' -module '//modpath//' -I '//modpath 
-    case('debug_pgfortran','debug_pgf90','debug_pgf95')  ! Portland Group F90/F95 compilers
-       fflags = ' '
-       mandatory=' -module '//modpath//' -I '//modpath 
-
-    case('release_flang')
-       fflags = ' '
-       mandatory=' -module '//modpath//' -I '//modpath 
-    case('debug_flang')
-       fflags = ' '
-       mandatory=' -module '//modpath//' -I '//modpath 
-
-    case('release_lfc')
-       fflags = ' '
-       mandatory=' -M '//modpath//' -I '//modpath 
-    case('debug_lfc')
-       fflags = ' '
-       mandatory=' -M '//modpath//' -I '//modpath 
-
-    case('release_nagfor')
-       fflags = ' &
-       & -O4&
-       & -coarray=single&
-       & -PIC&
-       &'
-       mandatory=' -mdir '//modpath//' -I '//modpath !
-    case('debug_nagfor')
-       fflags = '&
-       & -g&
-       & -C=all&
-       & -O0&
-       & -gline&
-       & -coarray=single&
-       & -PIC&
-       &'
-       mandatory=' -mdir '//modpath//' -I '//modpath !
-    case('release_crayftn')
-       fflags = ' '
-       mandatory=' -J '//modpath//' -I '//modpath 
-    case('debug_crayftn')
-       fflags = ' '
-       mandatory=' -J '//modpath//' -I '//modpath 
-
-    case('release_xlf90')
-       fflags = ' '
-       mandatory=' -qmoddir '//modpath//' -I '//modpath 
-    case('debug_xlf90')
-       fflags = ' '
-       mandatory=' -qmoddir '//modpath//' -I '//modpath 
+module fpm_compiler
+use fpm_model, only: fpm_model_t
+use fpm_filesystem, only: join_path, basename
+implicit none
+public :: get_module_flags
+public :: get_default_compile_flags
+public :: get_debug_compile_flags
+public :: get_release_compile_flags
+
+enum, bind(C)
+    enumerator :: &
+        id_unknown, &
+        id_gcc, &
+        id_f95, &
+        id_caf, &
+        id_intel_classic, &
+        id_intel_llvm, &
+        id_pgi, &
+        id_nvhpc, &
+        id_nag, &
+        id_flang, &
+        id_ibmxl, &
+        id_cray, &
+        id_lahey, &
+        id_lfortran
+end enum
+integer, parameter :: compiler_enum = kind(id_unknown)
+
+contains
+
+subroutine get_default_compile_flags(compiler, release, flags)
```

Comment:
I would add a comment/note here that this is the future point of extension if/when we want to have more profiles.
`release` will become a string (maybe an enumerator), and have a `select case` construct instead of an `if/else` construct.

Reply:
I actually expect this routine to vanish once we have compiler profiles available in the package manifest. The defaults would be encoded in the respective manifest type deserialization step.